batchflow/utils.py

```diff
@@ -371,3 +380,147 @@ def save_data_to(what, where, **kwargs):
             var[:] = item
         else:
             where[i] = item
+
+
+def get_notebook_path():
+    """ Return the full absolute path of the current jupyter notebook,
+    for example, `/path/path/path/My_notebook_title.ipynb`.
+    """
+    kernel_id = re.search('kernel-(.*).json',
+                          ipykernel.connect.get_connection_file()).group(1)
```

Comment:
What will kernel_id be, if this is run not from a notebook? I'd add a way to leave this function without iterating servers